src/main/java/org/javawebstack/orm/TableInfo.java

```diff
@@ -42,6 +42,18 @@
     public TableInfo(Class<? extends Model> model, ORMConfig config) throws ORMConfigurationException {
         this.config = config;
         this.modelClass = model;
+        if (model.getSuperclass() != Model.class) {
+            Class<? extends Model> superModel = (Class<? extends Model>) model.getSuperclass();
+            if (Modifier.isAbstract(superModel.getModifiers())) {
+                constructInfo(superModel);
+            } else {
+                throw new ORMConfigurationException("The parent model has to be abstract!");
+            }
+        }
+        constructInfo(model);
+    }
+
+    private void constructInfo (Class<? extends Model> model) throws ORMConfigurationException {
```

Comment:
When using the same method for analyzing the parent models and the actual model this should be split into 2 methods. One method for analyzing table properties (tableName, ...) and one for analyzing fields. With the current code it doesn't seem to be an issue because all table specific properties are fully reassigned but this could become an issue in the future.

Reply:
Agreed, but I think this refactoring is out of scope for now and should rather occur, when specific behaviour is needed.
I also thought about adding a merge method to the TableInfo class, to make it less complex.

Reply:
This can make serious trouble when changes are made to constructInfo in the future without taking care about this so I would rather split them up now. It shouldn't require much refactoring, just split it up into something like `void analyzeTable(Class<? extends Model> model)` and `void analyzeColumns(Class<? extends Model> model)` and only call analyzeColumns for the parent classes.